### PR TITLE
Trust Voice Lines, Trust: Lehko Habhoka, and Trust: Naja Salaheem

### PR DESCRIPTION
**Pull request by zach2good**
```
Lehko will only take 20 minutes to implement
               - zach2good (before stream)
```

I did a stream, got a lot done for Lehko Habhoka

This ended up needing a lot of tweaking behind the scenes:
- New spell targeting for `BEST_AGAINST_TARGET`
- New spell selection for `GetBestAgainstTargetWeakness`
- His 4 new TP moves
- Gambit changes for how we deal with in-melee ranged attacks
- Actually setting the `last_used` field on gambits, instead of on copies of gambits
- Gambit targeting for party-healing mobskills
- Setting the Trust's subjob based on the masters level, not their sub-level: `PTrust->SetSLevel(std::floor(PMaster->GetMLevel() / 2));`

image
_reeeeeeeeeeeeeeeeeeeeeeeeeee_

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by zach2good** [comment]
Do not merge, gonna fix and refactor some things while I'm in here

**Comment by zach2good** [comment]
Ok, this is tested and good to go 👍

**Comment by TeoTwawki** [comment]
Serious question regarding `[trust] Make Cornelia and Matsui-P summonable again`:

when retail had them they occupied the same slot. it seems retail means to keep using 1003 for "temporary" celebrity trusts. Should we really enable them at the same time or should we really have some kind of system where we pick which past temp trust fills the slot leaving it up to custom shenanigans to get them at once?

**Comment by zach2good** [comment]
Teo had the good ideas - removed Cornelia and Matsui-P from this PR, in favour of making them an optional module later. Leaving the commits for them as searchable documentation.

Safe to merge 👍

**Comment by kaincenteno** [comment]
purrrfect work on Naja and Lehko :smile:

**Review by TeoTwawki — APPROVED** [review]
looks good to me. 

I don't remember exactly why we avoided the tabling for the message offsets before but I guess we're better off this way so it doesn't give such a "magic number" appearance in each trust's home script.

**Review by claywar — APPROVED** [review]

File: src/map/mob_spell_container.cpp [hunk 1]
```diff
@@ -157,6 +157,73 @@ std::optional<SpellID> CMobSpellContainer::GetBestAvailable(SPELLFAMILY family)
...
+        {
+            choice = GetBestAvailable(SPELLFAMILY_DRAIN);
+            break;
+        }
+    }
+
+    // If all else fails, just cast the best you have!
+    return !choice ? GetBestAvailable(SPELLFAMILY_NONE) : choice;
+}
+
```
Comment (kaincenteno) [inline src/map/mob_spell_container.cpp:226]: will this make the mobs attack the players weakest elemental resistance or it only applies to attacks to mobs?

Comment (zach2good) [reply src/map/mob_spell_container.cpp:226]: This is currently only used for Trusts, we're trying to figure out if there are any actual mobs that need really smart AI or not

Comment (kaincenteno) [reply src/map/mob_spell_container.cpp:226]: awesome thanks =)

I'll try doing some campaign battle and intentionally lower my magic resistance and see if mobs tend to select that element more. would be good research ❤️

Comment (TeoTwawki) [reply src/map/mob_spell_container.cpp:226]: > This is currently only used for Trusts, we're trying to figure out if there are any actual mobs that need really smart AI or not

def were a couple back in 75 cap. my best example has sadly been dumbed down by SE and I don't know why.
https://ffxiclopedia.fandom.com/wiki/Phanduron_the_Condemned
> Rather high casting intelligence compared to most caster mobs.

they weren't talking about his INT stat. he'd hang onto stun till you were trying to do something he didn't want you doing, as just one example. If stun was on cooldown he'd sleep you instead just to interrupt you. Brought a healer? your healer is getting slept. Party sleepga'd. Shadows? he'll wait for you to just finish recasting then strip those immediately. That was all back before they moved his pos and added lv 80+ mobs to the zone. Now in modern ffxi he's a a dumb as bricks blm that selects spells at random even when it won't help him at all. :cry:

**Pull request merged**